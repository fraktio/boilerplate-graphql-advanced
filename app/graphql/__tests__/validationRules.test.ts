import {
  COMPLEXITY_COST,
  COMPLEXITY_LIMIT,
  createValidationRules,
  DEPTH_VALUE_LIMIT,
} from "~/graphql/validationRules";

describe("Graphql / schema", () => {
  it("createValidationRules", async () => {
    const rules = createValidationRules();

    expect(rules.length).toBe(2);
  });

  it("DEPTH_VALUE_LIMIT", async () => {
    expect(DEPTH_VALUE_LIMIT).toMatchSnapshot();
  });

  it("COMPLEXITY_LIMIT", async () => {
    expect(COMPLEXITY_LIMIT).toMatchSnapshot();
  });

  it("COMPLEXITY_COST", async () => {
    expect(COMPLEXITY_COST).toMatchSnapshot();
  });
});
