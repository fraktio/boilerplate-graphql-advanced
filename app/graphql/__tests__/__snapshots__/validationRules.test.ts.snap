// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphql / schema COMPLEXITY_COST 1`] = `
Object {
  "listFactor": 20,
  "objectCost": 10,
  "scalarCost": 2,
}
`;

exports[`Graphql / schema COMPLEXITY_LIMIT 1`] = `100000000000`;

exports[`Graphql / schema DEPTH_VALUE_LIMIT 1`] = `6`;
